### DISCUSSION TITLE
Enabling/disabling shortcuts based on opened file type

### DISCUSSION THREAD

**Discussion by gnakic** [JetBrains/ideavim#936] [General]
Would something like this be possible or is it somewhere in the roadmap?

I want to emulate some shortcuts from Doom Emacs that are populated in the
 "local leader" shortcut group. (`<Leader>m` in this case)
 
 So for example, say I open a Markdown file and I want to bold some text at cursor point, I would map that to be `<Leader>mb`, but if I open a Rust source file, there would be a completely different shortcut mapped at `<Leader>mb`. (or none at all if it wasn't mapped for the `.rs` file type)

**Comment by citizenmatt** [comment]
Unfortunately, this isn't possible right now. At some point, we'd like to support `autocmd` which would allow running vimscript in response to various events, and it would be nice to support file types here (although it would be somewhat different to Vim due to the language support coming from the IDE, not Vim's language plugins). Here's the issue for the request for `autocmd`, but there is currently no ETA for implementation: [VIM-1693](https://youtrack.jetbrains.com/issue/VIM-1693/Add-support-for-autocmd).
